src/main/drivers/pwm_output.c

```diff
@@ -289,9 +319,18 @@ void motorDevInit(const motorDevConfig_t *motorConfig, uint16_t idlePulse, uint8
 #endif
     }
 
+#ifdef USE_DSHOT_TELEMETRY
+    useDshotTelemetry = motorConfig->useDshotTelemetry;
+    if (useDshotTelemetry) {
+        useBurstDshot = false;
```

Comment:
Isn't useBurstDshot is actually enabled by default on all F4 boards?  Yet none that I've tried had issues running without burst. If we follow your suggested approach people with these boards could not try the feature at all, even if it works for OMNIBUSF4SD and NOX etc.

It might be better if we had a target based flag which would indicate if it's possible to run without burst. In that case the dshot_burst setting might not be offered at all since switching to off would render the quad unflyable.

In any case allowing users to try using telemetry and switching off burst when they do does not appear to be more problematic than offering the option to switch burst off by itself.

We can clearly document this and once we know which boards work move to the flag based approach I described?

Reply:
There are plenty of boards that have DMA resource conflicts restricting the use of some pins when burst Dshot isn't used. Speaking from experience in designing boards for Dshot before burst Dshot, there is barely any combination possible that has no DMA conflicts, so I would not expect any of the boards released after burst Dshot to even try and support running without it.

Disabling burst Dshot 'behind the scenes' without properly disabling it when the config is validated will leave many users with a broken and unflyable configuration. We could disable burst dshot in the config validation, but we don't have any way to detect if this is causing a DMA resource conflict, and leaving users with a non-working configuration is not a good approach to make them want to use a feature, so in my opinion, the better approach is to disable ERPM telemetry until the user has consciously disabled burst Dshot and verified that the function that they need are all still working.

Burst Dshot was enabled for all boards since there seemed to be no downside to it (until now that is). In general, the nightmare that Dshot is DMA wise without burst Dshot is not something that anybody would want to get back into, so I think the only good solution here will be to add burst Dshot support to ERPM telemetry.

Reply:
Ah - right. Yeah we can do it that way. We just have to tell users clearly that they first have to turn burst off before trying telemetry then. So that they are not confused when the feature automatically disables after they set it in the CLI.

Reply:
The 'automatic disable' is a pattern that we use quite a bit in config validation - e.g. if you try enabling conflicting features (say OneShot125 and 8 kHz PID loop), then one of the features will be reset after you save the config.
This can be dealt with in the documentation, and at least it makes spotting bug reports for this ('my feature X won't turn on) easy.
It isn't user friendly, but it's better than just accepting a broken configuration, and with everybody developing new features and nobody developing better configurator integration, this isn't going to become better any time soon. ;-)